packages/pipeline/src/types.ts

```diff
@@ -0,0 +1,85 @@
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+import { ReactNode } from 'react';
+
+export const Size = {
+  XSmall: 'xsmall',
+  Small: 'small',
+  Normal: 'normal',
+  Large: 'large',
+} as const;
+
+export type Size = typeof Size[keyof typeof Size];
+
+export interface StateForStyles {
+  hasHiddenStages: boolean;
+  size: Size;
+}
+
+export interface PipelineProps extends HTMLElementProps<'div', never> {
+  /**
+   * Content that will appear inside of the Pipeline component.
+   */
+  children: ReactNode;
+
+  /**
+   * Optional className prop to apply to Pipeline content container.
+   */
+  className?: string;
+
+  /**
+   * Alter the rendered size of the component
+   */
+  size: Size;
+
+  /**
+   * Renders the component with dark mode styles.
+   *
+   * @default false
+   */
+  darkMode?: boolean;
+}
+
+export interface StageProps extends HTMLElementProps<'li', never> {
+  /**
+   * Content that will appear inside of the Stage component.
+   **/
+  children: ReactNode;
+
+  /**
+   * ClassName applied to Stage content container.
+   **/
+  className?: string;
+
+  /**
+   * The DOM node to use as the root node for the intersectionObserver. Defaults to window when null or undefined.
+   **/
+  intersectionNode?: HTMLElement | null;
+
+  /**
+   * Alter the rendered size of the component. Inherited from the parent Pipeline component.
+   */
+  size?: Size;
```

Comment:
Was considering moving `size` and `intersectionNode` into the `PipelineContext` but the description of `intersectionNode` in the README -- "The DOM element to observe intersections with. When used with the `Pipeline` component, this prop is automatically set to the element rendered by the `Pipeline`" -- makes it seem like you can use `Stage` outside of `Pipeline`  which I don't think is great. Seeing as this is a `Major` change maybe I should remove this from `Stage` and add to  `PipelineContext` because we shouldn't have consumers be able to pass a `size` to both `<Pipeline>` and `<Stage>` and override the default intersectionNode by passing in a new `intersectionNode`